kconfig-sof-default.sh

```diff
@@ -10,7 +10,7 @@ if test -f "$FILE"; then
     COMMAND=$FILE;
 else
     # try if the script in a work directory without the -build path
-    CODE_DIR=${BUILD_DIR%-build}
+    CODE_DIR=source
```

Comment:
out of curiosity, who creates this link? is this ktest.pl or something else?

Reply:
@plbossart TBH I don't know, but it's always there :-) I just tested `make O=/tmp/test defconfig` and the link was already created.

Reply:
this does not work in all cases @lyakh 

the 'source' link is created by the 'make defconfig' command, so if you start from a blank directory make defconfig fails because you don't have a makefile in the first place.

You would first need to go to the source tree, and use the command you listed.

Reply:
@plbossart you're right, sorry, I didn't look at the script well enough, I didn't notice, that you can call it in an empty directory without an initial defconfig, that it calls defconfig itself. Well, we can drop this, or we can make a 3-stage guessing, or we can prohibit calling this in an empty directory. If we drop this, well, at least one person - me - will have to add this manually because that person routinely does `make O=...` :-) But that wouldn't be too wild, I'm fine either way.

Reply:
ok, let's drop this. We've both learned where the 'source' link comes from and we might use it later. for now the script isn't broken.